include/deal.II/fe/fe.h

```diff
@@ -149,6 +149,67 @@ template <int dim, int spacedim> class FESystem;
  * FiniteElement::system_to_block_index(). The number of blocks of a finite
  * element can be determined by FiniteElement::n_blocks().
  *
+ * To better illustrate these concepts, let's consider the following
+ * example of the multi-component system
+ * @code
+ * FESystem<dim> fe_basis(FE_Q<dim>(2), dim, FE_Q<dim>(1),1);
+ * @endcode
+ * with <code>dim=2</code>. The resulting finite element has 3 components:
+ * two that come from the quadratic element and one from the linear element.
+ * If, for example, this system were used to discretize a problem in fluid
+ * dynamics then one could think of the first two components representing a
+ * vector-valued velocity field whereas the last one corresponds to the scalar
+ * pressure field. Without degree-of-freedom (DoF) renumbering this finite element will
+ * produce the following distribution of local DoFs:
+ *
+ * @image html fe_system_example.png DoF indices
+ *
+ * Using the two functions FiniteElement::system_to_component_index() and
+ * FiniteElement::system_to_base_index() one can get the
+ * following information for each degree-of-freedom "i":
+ * @code
+ * const unsigned int component     = fe_basis.system_to_component_index(i).first;
+ * const unsigned int within_base   = fe_basis.system_to_component_index(i).second;
+ * const unsigned int base          = fe_basis.system_to_base_index(i).first.first;
+ * const unsigned int multiplicity  = fe_basis.system_to_base_index(i).first.second;
+ * const unsigned int within_base_  = fe_basis.system_to_base_index(i).second; // same as above
```

Comment:
Sorry, I thought that I wrote a comment on this in my review: Should we add to this example what the overlap between `system_to_block_index` and `system_to_component_index` is? If so, then we also need to mention the former in the comment above the code.

Reply:
Maybe leave it for another day (one can extend the test to print extra data in the table and adjust explanation)? 

Personally, I am not sure what `system_to_block_index` will return here. I am also puzzled why it returns a pair of `std::pair<unsigned int,types::global_dof_index>` instead of `std::pair<unsigned int,unsigned int>`; The latter is what I expect based on the description:

> The vector block and the index inside the block for this shape function.

Reply:
Ok, fair enough!

Reply:
Block indices are complementary to components and bases, but they are not widely used in the library and generally poorly defined and documented. I think it's fair to leave them out of this patch. (The original idea of blocks was to identify "indivisible" objects. For example, an RT element has dim components, but it can't be taken apart. So it's counted as one "block". an `FESystem(FE_RT(), 2)` would have one base, two blocks, and 2*dim components.)